src/main/java/codeu/model/data/Message.java

```diff
@@ -63,6 +71,48 @@ public String getContent() {
     return content;
   }
 
+  /** Finds picture links that end in jpg and add to the end of the message the
+   * html rendered picture (link picture, opens to picture in another window)
+   * with maximum picture width at 500 */
+  public String parseImages(String message) { 
+    int startIndex = 0;
+    int linkStartIndex;
+    int spaceIndex;
+    int newlineIndex;
+    int endWordIndex;
+    List<String> links = new ArrayList<String>(); 
+
+    while (startIndex < message.length()) {
```

Comment:
Consider using regular expressions to look for this match instead.  You'd want to match something like.

"http(s)?://\S+\.jpg\s"

Roughly translating into:

Match http.  Optinally match the "s" for https.  Look for a string of non-space characters, follow by ".jpg", followed by a whitespace"

This will fail for matching messages where the URL ends the message since the trailing whitespace will never get matched.  A simple fix for this is to append a space to the end of the message before parsing.

Reply:
Feel free to ping if you run into problems implementing regular expressions.  They're super useful, but also somewhat steep learning curve.

Reply:
Index of does not seem to support regular expressions like some of the other string functions, and there doesn't really seem to be any other string functions that will return a string or the location of a substring using regular expressions